spec/abci++/abci++_properties_001_draft.md

```diff
@@ -0,0 +1,689 @@
+
+Property 6 constrains the creation and handling of vote extensions in a similar way as Property 3
+contrains the creation and handling of proposed blocks.
+Property 6 ensures that extensions created by correct processes _always_ pass the `VerifyVoteExtension`
+checks performed by correct processes receiving those extensions.
+However, if there is a (deterministic) bug in `ExtendVote` or `VerifyVoteExtension` (or in both),
+we will face the same liveness issues as described for Property 3, as Precommit messages with invalid vote
+extensions will be discarded.
+
+* Property 7 [`VerifyVoteExtension`, determinism-1]: For any correct process $p$,
+  and any arbitrary vote extension $e$, and any arbitrary block $w$,
+  if $p$'s (resp. $q$'s) Tendermint calls `RequestVerifyVoteExtension` on $e$ and $w$ at height $h$,
+  then $p$'s Application's acceptance or rejection exclusively depends on $e$, $w$ and $s_{p,h-1}$.
+
+* Property 8 [`VerifyVoteExtension`, determinism-2]: For any two correct processes $p$ and $q$,
+  and any arbitrary vote extension $e$, and any arbitrary block $w$,
+  if $p$'s (resp. $q$'s) Tendermint calls `RequestVerifyVoteExtension` on $e$ and $w$ at height $h$,
+  then $p$'s Application accepts $e$ if and only if $q$'s Application accepts $e$.
+  Note that this property follows from Property 7 and the Agreement property of consensus.
+
+Properties 7 and 8 ensure that the validation of vote extensions will be deterministic at all
+correct processes.
+Properties 7 and 8 protect against arbitrary vote extension data from Byzantine processes
+similarly to Properties 4 and 5 and proposed blocks.
+Properties 7 and 8 can be violated by a bug inducing non-determinism in `ExtendVote` or
+`VerifyVoteExtension`. In this case liveness can be compromised.
+Extra care should be put in the implementation of `ExtendVote` and `VerifyVoteExtension` and,
+as a general rule, `VerifyVoteExtension` _should_ always accept the vote extension.
+
+* Property 9 [_all_, read-only]: $p$'s calls to `RequestPrepareProposal`, `RequestProcessProposal`,
+  `RequestExtendVote`, and `RequestVerifyVoteExtension` at height $h$ do not modify $s_{p,h-1}$.
+
+* Property 10 [`ExtendVote`, `FinalizeBlock`, non-dependency]: for any correct process $p$,
+and any vote extension $e$ that $q$ received at height $h$, the computation of
+$s{p,h}$ does not depend on $e$.
+
+The call to correct process $p$'s `RequestFinalizeBlock` at height $h$, with block $v_{p,h}$
+passed as parameter, creates state $s_{p,h}$.
+Additionally, $p$'s `FinalizeBlock` creates a set of transaction results $T_{p,h}$.
+
+>**TODO** I have left out all the "events" as they don't have any impact in safety or liveness
+>(same for consensus params, and validator set)
+
+* Property 11 [`FinalizeBlock`, determinism-1]: For any correct process $p$,
+  the contents of $s_{p,h}$ exclusively depend on $s_{p,h}$ and $v_{p,h}$.
```

Comment:
I guess this should be rephrased. I am not sure about "contents". Also there is a wrong index in "$s_{p,h}$ exclusively depend on $s_{p,h}$"

Reply:
Yes. Thanks for catching. Fix coming with the commit I am about to push